### PR TITLE
feat: refactor getTxById to receive txId by path param

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- pass txId by path param, it will simplify the endpoint usage and comply with HTTP REST spec
- change route to wallet/transactions/{txId}

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -496,10 +496,14 @@ functions:
     handler: src/api/txById.get
     events:
       - http:
-          path: wallet/getTxById
+          path: wallet/txById/{txId}
```
Comment (andreabadesso) [inline serverless.yml]: What do you think about refactoring it to be something line:

`wallet/transactions/{txId}`

I think it looks a lot more `restful`

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: serverless.yml (hunk 1 above)
Comment (alexruzenhack) [reply serverless.yml]: @andreabadesso, you're right. I didn't use `/transactions` because I was afraid to use it, but it is indeed better.

**Review by andreabadesso — APPROVED** [review]

File: serverless.yml (hunk 1 above)
Comment (andreabadesso) [reply serverless.yml]: @alexruzenhack thanks!

Don't forget to update the acceptance criteria

File: src/api/txById.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ const mysql = getDbConnection();
 
 class TxByIdValidator {
   static readonly bodySchema = Joi.object({
-    txId: Joi.string().required(),
+    txId: Joi.string().min(1).required(),
```
Comment (andreabadesso) [inline src/api/txById.ts]: I guess we can use `.min(32).max(32)` here since `txId` will always be a `sha256` hash

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: src/api/txById.ts (hunk 2 above)
Comment (alexruzenhack) [reply src/api/txById.ts]: Are you sure it is 32 characters? Doesn't it 64?

Comment (andreabadesso) [reply src/api/txById.ts]: Oops, it's in hexadecimal string format, so yes, 64 bytes

Sorry

File: src/api/txById.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ const mysql = getDbConnection();
 
 class TxByIdValidator {
   static readonly bodySchema = Joi.object({
-    txId: Joi.string().required(),
+    txId: Joi.string().min(32).max(32).required(),
```
Comment (tuliomir) [inline src/api/txById.ts]: From previous conversations I see that this should be `min(64).max(64)`

**Review by tuliomir — APPROVED** [review]

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**